lib/ui/layout.js

```diff
@@ -1,101 +1,139 @@
 'use babel'
 
-export function queryStandardLayout () {
-  let msg = atom.notifications.addInfo("Use Standard Layout?", {
-    buttons: [{text: "Yes", onDidClick: () => {
-      standard()
-      msg.dismiss()
-      atom.config.set('julia-client.uiOptions.useStandardLayout', true)
-      atom.config.set('julia-client.firstBoot', false)
-    }}, {text: "No", onDidClick: () => {
-      msg.dismiss()
-      atom.config.set('julia-client.uiOptions.useStandardLayout', false)
-      atom.config.set('julia-client.firstBoot', false)
-    }}],
-    description: `Opens various Julia specific panes in a default layout.
-    Use the \`Julia: Standard Layout\` command to reset the layout at later point in time.`,
-    dismissable: true
-  })
+// @NOTE: Sort panes by importance for `openPanesHelper`
+const repl = () => {
+  return require('../runtime').console2
+}
+const workspace = () => {
+  return require('../runtime').workspace
+}
+const documentation = () => {
+  return require('../ui').docpane
+}
+const plotPane = () => {
+  return require('../runtime').plots
+}
+const debuggerPane = () => {
+  return require('../runtime').debugger
+}
+const linter = () => {
+  return require('../runtime').linter
 }
 
-function getPanes() {
-  if (atom.workspace.getBottomDock) {
-    return atom.workspace.getPanes().slice(0, -3)
-  } else {
-    return atom.workspace.getPanes()
-  }
+function specifiedPanes () {
+  const panes = []
+  if (atom.config.get('julia-client.uiOptions.layouts.defaultPanes.console')) panes.push(repl)
+  if (atom.config.get('julia-client.uiOptions.layouts.defaultPanes.workspace')) panes.push(workspace)
+  if (atom.config.get('julia-client.uiOptions.layouts.defaultPanes.documentation')) panes.push(documentation)
+  if (atom.config.get('julia-client.uiOptions.layouts.defaultPanes.plotPane')) panes.push(plotPane)
+  if (atom.config.get('julia-client.uiOptions.layouts.defaultPanes.debuggerPane')) panes.push(debuggerPane)
+  if (atom.config.get('julia-client.uiOptions.layouts.defaultPanes.linter')) panes.push(linter)
+
+  return panes
 }
 
-export function isCustomLayout() {
-  return getPanes().length > 1
+function closePromises () {
+  // Close only specified panes, i.e.: non-specified panes won't layouted
+  const panes = specifiedPanes()
```

Comment:
I think it would be best to flatten all items in the workspace center into one pane and then only closing Juno pane items that are in the workspace center. After that you can focus the one remaining pane in the workspace center and start splitting stuff from there.

But the current implementation is fine too, so no need to change that if you don't want to.

Reply:
In the current implementation, I does that too.
The function `bundlePanes` flattens all the items in each container, cause now we have an option to open Juno panes outside of `workspaceCenter` (and without this, say, right dock can be funky state after the command as well like you showed yesterday)

Reply:
Right, but it's much less likely to have splits in docks (at least for me).
We could also specialcase the no-splits case and don't close/reopen stuff that's in the right place already.

Reply:
> We could also specialcase the no-splits case and don't close/reopen stuff that's in the right place already.

IMO we better to close/reopen even with `no-split`, cause a pane with `split === 'no split'` can be outside of the `defaultLocation` and it should be moved into the right location.

Reply:
Sure, we'd need to check that each pane item is in the correct place (i.e. workspace center or dock), that there are no splits in that pane, and that the user doesn't want any splits for those items. But that's probably not worth the effort.

Reply:
Ah I see. Agree with you that I also wonder it's worth doing. So let this go for now.